lib/chef/provider/user/solaris.rb

```diff
@@ -46,19 +46,17 @@ def manage_user
         end
 
         def check_lock
-          shadow_line = shell_out!("getent", "shadow", new_resource.username).stdout.strip rescue nil
-
-          # if the command fails we return nil, this can happen if the user
-          # in question doesn't exist
-          return nil if shadow_line.nil?
-
-          # convert "dave:NP:16507::::::\n" to "NP"
-          fields = shadow_line.split(":")
+          @locked = nil
+          ::File.readlines(@password_file).each do |entry|
```

Comment:
My Friday night brain reads that in Ethel Mermen's voice singing "Everything's coming up nil"...